test-data/unit/sql-dataframe.test

```diff
@@ -21,6 +21,77 @@ df.sample(withReplacement=False) # E: No overload variant of "sample" of "DataFr
 
 [out]
 
+[case selectColumns]
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col
+
+spark = SparkSession.builder.getOrCreate()
+
+data = [('Alice', 1)]
+df = spark.createDataFrame(data, schema="name str, age int")
+
+df.select(["name", "age"])
+df.select([col("name"), col("age")])
+
+df.select(["name", col("age")])  # E: Argument 1 to "select" of "DataFrame" has incompatible type "List[object]"; expected "Union[List[Column], List[str]]"
+
+[out]
+
+[case groupBy]
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col
+
+spark = SparkSession.builder.getOrCreate()
+
+data = [('Alice', 1)]
+df = spark.createDataFrame(data, schema="name str, age int")
+
+df.groupBy(["name", "age"])
+df.groupby(["name", "age"])
+df.groupBy([col("name"), col("age")])
+df.groupby([col("name"), col("age")])
+
+
+df.groupBy(["name", col("age")])  # E: Argument 1 to "groupBy" of "DataFrame" has incompatible type "List[object]"; expected "Union[List[Column], List[str]]"
```

Comment:
I guess this is an acceptable trade-off.

Reply:
I hope so. pyspark considers this mixture legal, but I have seen many `List[str]` or `List[Column]` and never such a hybrid.

It's a shame though that we can't get everything...